cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -0,0 +1,143 @@
+ ** <<cypher-deprecations-additions-removals-3.1, Version 3.1>>
+ ** <<cypher-deprecations-additions-removals-3.2, Version 3.2>>
+ ** <<cypher-deprecations-additions-removals-3.3, Version 3.3>>
+* <<cypher-compatibility, Compatibility>>
+ ** <<cypher-start-compatibility, Accessing entities by id via START>>
+* <<cypher-versions, Supported language versions>>
+
+[abstract]
+--
+Cypher is a language that is constantly evolving.
+New features get added to the language continuously, and occasionally, some features become deprecated and are subsequently removed.
+--
+
+[[cypher-deprecations-additions-removals]]
+== Removals, deprecations, additions and extensions
+
+The following tables lists all the features which have been removed, deprecated, added or extended in Cypher.
+Replacement syntax for deprecated and removed features are also indicated.
+
+[[cypher-deprecations-additions-removals-3.0]]
+=== Version 3.0
+[options="header"]
+|===
+| Feature          | Type | Change | Details
+| `has()`  | Function  | Removed | Replaced by <<functions-exists, exists()>>
+| `str()`  | Function  | Removed | Replaced by <<functions-tostring, toString()>>
+| `{parameter}` | Syntax | Deprecated | Replaced by <<cypher-parameters, $parameter>>
+| <<functions-properties, properties()>>  | Function  | Added  |
+| <<query-call,CALL [\...YIELD]>>   | Clause  | Added  |
+| <<functions-point-cartesian,point() - cartesian 2D>> | Function | Added |
+| <<functions-point,point() - WGS 84>> | Function | Added |
+| <<functions-distance,distance()>> | Function | Added |
+| <<procedures, User-defined procedures>> | Functionality | Added |
+|  <<functions-tostring, toString()>>   | Function  | Extended | Now also allows Boolean values as input
+|===
+
+[[cypher-deprecations-additions-removals-3.1]]
+=== Version 3.1
+[options="header"]
+|===
+| Feature          | Type | Change | Details
+| `rels()`   | Function  | Deprecated | Replaced by <<functions-relationships, relationships()>>
+| `toInt()`   | Function  | Deprecated | Replaced by <<functions-tointeger, toInteger()>>
+| `lower()`   | Function  | Deprecated | Replaced by <<functions-tolower, toLower()>>
+| `upper()`   | Function  | Deprecated | Replaced by <<functions-toupper, toUpper()>>
+| <<functions-toboolean,toBoolean()>> | Function | Added |
+| <<cypher-map-projection, Map projection>> | Syntax | Added |
+| <<cypher-pattern-comprehension, Pattern comprehension>> | Syntax | Added |
+| <<user-defined-functions, User-defined functions>> | Functionality | Added |
+| <<query-call, CALL\...YIELD\...WHERE>>   | Clause  | Extended  | Records returned by `YIELD` may be filtered further using `WHERE`
+|===
+
+[[cypher-deprecations-additions-removals-3.2]]
+=== Version 3.2
+[options="header"]
+|===
+| Feature          | Type | Change | Details
+| `CYPHER planner=rule` (Rule planner)    | Functionality | Removed | All queries now use the cost planner. Any query prepended thus will fall back to using Cypher 3.1.
```

Comment:
Not sure I would write "removed" here. If you are using Neo4j 3.2 and specify "planner=rule", that query will still work and return the expected results. I would say deprecated here. Same goes for the next element, `CREATE UNIQUE`.

Reply:
This is an odd case. For Neo4j 3.2 we have removed the rule planner, but since we support Cypher 3.1, and that bundles its own rule planner, we can still get it that way. So the question is, does the fact that we coded a convenience of having 'Cypher planner=rule' automatically mean 'Cypher 3.1 planner=rule' mean that we should not use the word 'removed' and instead use the word 'deprecated'?

Reply:
We could argue that if the title of this table `Version 3.2` refers to Cypher version, then these features are indeed _removed_ and not merely deprecated. If we are refering to Neo4j the product, then they are deprecated because we automatically switch to and older Cypher.

My vote is for keeping the word '_removed_'.

Reply:
Yup, I was flip-flopping on this one a lot. I settled on `Removed` for the reason Craig mentioned. So, will keep as is, and hopefully the wording (all about how the query falls back yada yada) is sufficient to make this distinction clear to readers.

Reply:
👍